src/lm_buddy/jobs/_entrypoints/ragas.py

```diff
@@ -0,0 +1,105 @@
+from pathlib import Path
+
+from datasets import load_dataset
+from langchain_community.embeddings import HuggingFaceEmbeddings
+from langchain_openai import ChatOpenAI
+from lmbuddy.integrations.huggingface import HuggingFaceAssetLoader
+from ragas import evaluate
+from ragas.metrics import answer_relevancy, context_precision, context_recall, faithfulness
+
+from lm_buddy.integrations.wandb import artifact_loader
+from lm_buddy.integrations.wandb.artifact_config import WandbArtifactConfig
+from lm_buddy.integrations.wandb.artifact_utils import (
+    ArtifactType,
+    build_directory_artifact,
+    default_artifact_name,
+)
+from lm_buddy.integrations.wandb.run_utils import wandb_init_from_config
+from lm_buddy.jobs.common import EvaluationResult, LMBuddyJobType
+from lm_buddy.jobs.configs import RagasJobConfig
+
+RAGAS_METRICS_MAP = {
+    "faithfulness": faithfulness,
+    "answer_relevancy": answer_relevancy,
+    "context_recall": context_recall,
+    "context_precision": context_precision,
+}
+
+
+def run_eval(config: RagasJobConfig, artifact_loader: artifact_loader) -> Path:
+    # load dataset from W&B artifact
+    hf_loader = HuggingFaceAssetLoader(artifact_loader)
+    evaluation_dataset = hf_loader.load_dataset(config.dataset)
+
+    # ragas custom model args
+    ragas_args = {}
+
+    # load embedding model
+    ragas_args["embeddings"] = HuggingFaceEmbeddings(model_name=config.evaluation.embedding_model)
+
+    # configure ragas to point to vllm instance for generation
+    ragas_args["llm"] = ChatOpenAI(
+        model=config.ragas_inference_server.engine,
+        openai_api_key=config.evaluation.openai_api_key,
```

Comment:
If the env variable approach works with langchain's implementation of the openai client, you might not need this line